testing/integration/package_version_test.go

```diff
@@ -50,6 +55,107 @@ func TestPackageVersion(t *testing.T) {
 	t.Run("remove package versions file and test version again", testAfterRemovingPkgVersionFiles(ctx, f))
 }
 
+func TestComponentBuildHashInDiagnostics(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Fleet,
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+	ctx := context.Background()
+
+	f, err := atesting.NewFixture(
+		t,
+		define.Version(),
+		// TODO(AndersonQ) replace with ArtifactFetcher as soon as the beats
+		// with those changes are available in the API
+		atesting.WithFetcher(atesting.LocalFetcher("../../build/distributions/")),
+	)
+	require.NoError(t, err, "could not create new fixture")
+
+	err = f.Prepare(ctx)
+	require.NoError(t, err, "could not prepare fixture")
+
+	// Set package version
+	pkgVersionFiles := findPkgVersionFiles(t, f.WorkDir())
+	pkgVersion := "8.13.0+build20131123"
+	for _, pkgVerFile := range pkgVersionFiles {
+		err := os.WriteFile(pkgVerFile, []byte(pkgVersion), 0o644)
+		require.NoError(t, err)
+	}
```

Comment:
I would prefer we prepare a dedicated package with a specific version instead of hacking through the version files changing the values. I am not sure this will work once the version is part of the agent path...

Reply:
ok, I agree, but other tests are doing the same. Therefore it could be a followup PR changing all the test. Also to do so it'd require to build another agent for the test, changing the step on buildkite as well. What ins't quite the scope of this PR